suites/portal/indexingPageTest.js

```diff
@@ -0,0 +1,136 @@
+      console.log(e);
+      if (i === nAttempts - 1) {
+        throw e;
+      }
+    }
+  }
+}
+
+// Scenario #1 - Login and navigate to the indexing page and upload dummy manifest
+Scenario('Navigate to the indexing page and upload a test manifest @indexing', async (I, indexing, home) => {
+  home.do.goToHomepage();
+  home.complete.login();
+  indexing.do.goToIndexingPage();
+  I.waitForElement({ css: '.indexing-page' }, 10);
+  I.click('.index-flow-form'); // after clicking open window file upload dialog
+  await I.attachFile('input[type=\'file\']', `manifest_${I.cache.UNIQUE_NUM}.tsv`);
+  I.click({ xpath: 'xpath: //button[contains(text(), \'Index Files\')]' });
+
+  await checkPod('manifest-indexing');
+
+  const nAttempts = 5;
+  for (let i = 0; i < nAttempts; i += 1) {
+    console.log(`Looking up the indexd record... - attempt ${i}`);
+    const indexdRecordRes = await I.sendGetRequest(
+      `/index/${testGUID}`,
+    ).then((res) => new Gen3Response(res));
+
+    if (indexdRecordRes.data.hashes) {
+      expect(indexdRecordRes.data.hashes.md5).to.equal(testHash);
+      break;
+    } else {
+      console.log(`WARN: The indexd record has not been created yet... - attempt ${i}`);
+      await sleepMS(5000);
+      if (i === nAttempts - 1) {
+        const manifestIndexingLogs = bash.runCommand('gen3 job logs manifest-indexing');
+        console.log(`manifest-indexing logs: ${manifestIndexingLogs}`);
+        throw new Error(`ERROR: The manifest indexing operation failed. Response: ${indexdRecordRes.data}`);
+      }
+    }
+  }
+}).retry(2);
+
+// Scenario #2 - Login and navigate to the indexing page and download a full indexd manifest
+Scenario('Navigate to the indexing page and download a full indexd manifest @indexing', async (I, indexing, home) => {
+  home.do.goToHomepage();
+  home.complete.login();
+  indexing.do.goToIndexingPage();
+  I.waitForElement({ css: '.indexing-page' }, 10);
+  I.click({ xpath: 'xpath: //button[contains(text(), \'Download\')]' });
+
+  await checkPod('indexd-manifest');
+
+  const waitingThreshold = 60;
+  console.log('Waiting for Green status DONE to show up on the page...');
+  await I.waitForElement({ css: '.index-files-green-label' }, waitingThreshold);
+
+  await I.click({ xpath: 'xpath: //button[contains(text(), \'Download Manifest\')]' });
+  // TODO: Inject the react -> state: { downloadManifestLink } url into an anchor tag
+  const manifestDownloadUrl = await I.grabValueFrom({ xpath: 'xpath: //button[contains(text(), \'Download Manifest\')]' });
+  console.log(`### Manifest download url: ${manifestDownloadUrl}`);
+  const getManifestRes = await I.sendGetRequest(
+    manifestDownloadUrl.toString(),
+  ).then((res) => new Gen3Response(res));
+  console.log(`### downloadOuput: ${JSON.stringify(getManifestRes)}`);
+
+  const testManifestData = contentsOfTestManifest.split('\n')[1];
+  const downloadedManifestData = getManifestRes.body;
+
+  expect(
+    downloadedManifestData,
+  ).to.include(
```

Comment:
if the BeforeSuite is deleting all indexd records, this test should still pass if you use the stricter "equal" instead of "include" yeah?

Reply:
The 1st Scenario creates the indexd record and the 2nd Scenario downloads the manifest and makes sure the same data is found (GUID, md5 hash, size, url, etc.).
CodeceptJS `BeforeSuite` runs before all scenarios so that doesn't impact the 2nd test scenario.
There's a separate `Before` function that runs before every Scenario (which behaves similarly to Python's unittest `SetUp()` function), that one would totally mess this up, but that is not used in this script.

Reply:
would the test pass if you used "equal" instead of "include" ? i feel like a stricter condition is preferable ~

Reply:
are there other records or might there be duplicates?

Reply:
I'm using include because the test `manifest.tsv` file also becomes an indexd record (but I only care about the actual record that is created through the sower job, not the one that is created by `ssjdispatcher`).
The `BeforeSuite` should delete 'em all so we don't expect other records.
This test will definitely pass if there are duplicates, that is a good point.
Is it realistic to expect the sower job to create duplicates though??